src/averageAges.js

```diff
@@ -1,64 +1,52 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const totalAge = (array) => {
+  return array.reduce(
+    (acc, per) => acc + (per.died - per.born), 0);
+};
+
+const conditionCheck = (firstCondition, secondCondition) => {
+  return firstCondition === secondCondition;
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const newPeople = century ? people.filter(
+    person => conditionCheck(person.sex, 'm')
+    && Math.ceil(person.died / 100) === century)
+    : people.filter(person => conditionCheck(person.sex, 'm'));
+
+  return totalAge(newPeople) / newPeople.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const woman = withChildren
+    ? people.filter(person => conditionCheck(person.sex, 'f')
+    && people.some(mother => conditionCheck(person.name, mother.mother)))
+    : people.filter(person => conditionCheck(person.sex, 'f'));
+
+  return totalAge(woman) / woman.length;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = onlyWithSon
+    ? people.filter(person =>
+      people.some(child => conditionCheck(child.name, person.mother))
+    && conditionCheck(person.sex, 'm'))
+    : people.filter(person =>
+      people.some(child => conditionCheck(child.name, person.mother)));
```

Comment:
let's improve your solution 
```suggestion
  const children = people.filter(person => {
   const isMalePerson = person.sex === 'm';
   if (onlyWithSon) {
    return people.some(child => child.name === person.mother) && isMalePerson;
   }
   return isMalePerson;
  })
```